src/dmd/dstruct.d

```diff
@@ -136,7 +136,7 @@ extern (C++) void semanticTypeInfo(Scope* sc, Type t)
                 // https://issues.dlang.org/show_bug.cgi?id=15149
                 // if the typeid operand type comes from a
                 // result of auto function, it may be yet speculative.
-                unSpeculative(sc, sd);
+                // unSpeculative(sc, sd);
```

Comment:
Why leave it?

Reply:
Because I'm not sure that it works for all cases. If a bug report appears at some point, the person that will fix will have some hints to some resources. I've seen this pattern in some parts of the compiler.

Reply:
There's the Git history as well.